### DISCUSSION TITLE
Bugs in simulator 1.2.0

### DISCUSSION THREAD

**Discussion by kartun83** [zepp-health/zeppos-forum#166] [Developer Feedback]
There is a critical bug in simulator build 1.2.0 (latest to date), which prevents running any app in sim.

To be short:
in device.js
`const { device = [] } = lodash_1.find(cacheList, { id: platform }) || {};`

should be replaced with
`const { device = [] } = lodash_1.find(cacheList, { id: platform.id }) || {};`

also in create-simulator.js
```
function getPlatformDevice(platform) {
    const startPath = path_1.default.resolve(process.env.EMULATOR_CACHE_PATH || '', `${platform}`);
    return startPath;
}
```
should be patched to reflect that platform is an object with properties :)
```
function getPlatformDevice(platform) {
    const startPath = path_1.default.resolve(process.env.EMULATOR_CACHE_PATH || '', `${platform.id}`);
    return startPath;
}
```

possibly there are other parts that needs to be changed, but at least with this modifications simulator is able to run programs deployed with zeus.
---
Also suggestion:
add some flag to simulator so it would launch qemu without sudo.

**Comment by MyoungXUE** [comment]
Thank you for your feedback, I will feedback this issue to the relevant colleagues.

**Reply by kartun83** [reply]
Slight correction - this issue is reproducible if no ROM selected in dropdown list. Otherwise it seems to be working correctly.
